frontend/packages/knative-plugin/src/components/traffic-splitting/TrafficSplittingModal.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { FormikProps, FormikValues } from 'formik';
+import {
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory/modal';
+import { TextInputTypes } from '@patternfly/react-core';
+import {
+  MultiColumnField,
+  InputField,
+  DropdownField,
+} from '@console/dev-console/src/components/formik-fields';
+
+export interface TrafficSplittingModalProps {
+  revisionItems: any;
+  errorMessage?: string;
+  onCancel: () => void;
+}
+
+type Props = FormikProps<FormikValues> & TrafficSplittingModalProps;
+
+const TrafficSplittingModal: React.FC<Props> = ({
+  errorMessage,
+  revisionItems,
+  handleSubmit,
+  isSubmitting,
+  onCancel,
+}) => {
+  const handleSave = (e) => {
```

Comment:
No need to create this function for submit. You can directly pass `onSubmit={handleSubmit}` to the form. Formik adds `e.preventDefault` to `handleSubmit`